src/ergtable.h

```diff
@@ -0,0 +1,188 @@
+#ifndef ERGTABLE_H
+#define ERGTABLE_H
+
+#include <QList>
+#include <QSettings>
+#include <QObject>
+#include <QDebug>
+#include <QDateTime>>
+#include "qzsettings.h"
+
+struct ergDataPoint {
+    uint16_t cadence = 0; // RPM
+    uint16_t wattage = 0; // Watts
+    uint16_t resistance = 0; // Some unit
+
+    ergDataPoint() = default;
+
+    ergDataPoint(uint16_t c, uint16_t w, uint16_t r) : cadence(c), wattage(w), resistance(r) {}
+};
+
+Q_DECLARE_METATYPE(ergDataPoint)
+
+class ergTable : public QObject {
+    Q_OBJECT
+
+public:
+    ergTable(QObject *parent = nullptr) : QObject(parent) {
+        loadSettings();
+    }
+
+    ~ergTable() {
+        saveSettings();
+    }
+
+    void collectData(uint16_t cadence, uint16_t wattage, uint16_t resistance, bool ignoreResistanceTiming = false) {
+        if(resistance != lastResistanceValue) {
+            qDebug() << "resistance changed";
+            lastResistanceTime = QDateTime::currentDateTime();
+            lastResistanceValue = resistance;
+        }
+        if(lastResistanceTime.msecsTo(QDateTime::currentDateTime()) < 1000 && ignoreResistanceTiming == false) {
```

Comment:
In defense of the RP Pico, you could do the boolean check first to determine if the (admittedly tiny) effort to get the time do the calculation is actually necessary.

Reply:
that boolean is only for gtests so it's useless on runtime